### ISSUE TITLE
MapboxGL fails to turn off its own locationManager

### ISSUE THREAD

**Issue by christocracy** [transistorsoft/rn-background-geolocation-demo#1]
I've [pointed out](https://github.com/mapbox/react-native-mapbox-gl/issues/77#issuecomment-129657042) to the folks at MapBoxGL that their component fails to shut down their `CLLocationManager` from updating location when app goes to background.

While their component only asks for foreground location-updates, because `react-native-background-geolocation` asks for background-permission, theirs too will continue to run (at full GPS power) in background while mine tries to go to sleep in a stationary-geofence.  Since theirs keep running, the entire app is prevented from suspending, rapidly draining the battery (and getting your phone pretty hot).

I sent a [pull-request](https://github.com/mapbox/react-native-mapbox-gl/pull/78) to fix the problem but the MapBox people have requested teh fix be applied further upstream, at [MGLMapView.mm](https://github.com/mapbox/mapbox-gl-native/blob/master/platform/ios/MGLMapView.mm) (the actual `CLLocationManagerDelegate`, isolated from react-native)

tl;dr Until I send a PR to the people at [mapbox-gl-native](https://github.com/mapbox/mapbox-gl-native), this demo will be horrible on battery-usage.  This is not related to `react-native-background-geolocation`.

**Comment by jlebensold** [comment]
@christocracy what do you recommend as a mapping library for ios? Would it be possible to pull down a branch of this demo with a working map for iOS (even if you've forked the upstream dependencies)?

**Comment by christocracy** [comment]
I've currently got Mapbox-gl running again.  I hope to push an update this week.

I'm hoping [my issue with mapbox-gl](https://github.com/mapbox/mapbox-gl-native/pull/2120) is solved since my first attempts months ago, where mapbox-gl doesn't take care to turn off their `CLLocationManager` instance in the background, was solved with iOS 9 (which requires a `CLLocationManager` instance to explicitly declare [allowsBackgroundUpdates](https://developer.apple.com/library/ios/documentation/CoreLocation/Reference/CLLocationManager_Class/#//apple_ref/occ/instp/CLLocationManager/allowsBackgroundLocationUpdates)

This is very important, since my plugin declares the `UIBackgroundMode: "location"`, if some other 3rd party code is also running a location-updates (such as MapBoxGL), even if their library didn't declare this background-mode, their `CLLocationManager` can continue running even though my library wisely turns its location-updates off when the device is determined to be stationary.

So, with iOS9, the problems associated with MapBoxGL's naivete *should* be solved.

**Comment by jlebensold** [comment]
ok sounds good!
